base/applications/rapps/unattended.cpp

```diff
@@ -68,6 +108,127 @@ BOOL UseCmdParameters(LPWSTR lpCmdLine)
+    {
+        return TRUE;
+    }
 
-    return FALSE;
+    ConPrintf(StdOut, (LPWSTR)L"%s (%s)\n", (LPCWSTR)(Info->m_szName), (LPCWSTR)(Info->m_szPkgName));
+    return TRUE;
+}
+
+BOOL HandleFindCommand(LPWSTR szCommand, int argcLeft, LPWSTR *argvLeft)
+{
+    if (argcLeft < 1)
+    {
+        ConResMsgPrintf(StdOut, NULL, IDS_CMD_NEED_PARAMS, szCommand);
+        ConPrintf(StdOut, (LPWSTR)L"\n");
+        return FALSE;
+    }
+
+    CAvailableApps apps;
+    apps.UpdateAppsDB();
+
+    for (int i = 0; i < argcLeft; i++)
+    {
+        ConResMsgPrintf(StdOut, NULL, IDS_CMD_FIND_RESULT_FOR, argvLeft[i]);
+        ConPrintf(StdOut, (LPWSTR)L"\n");
+        apps.Enum(ENUM_ALL_AVAILABLE, CmdFindAppEnum, argvLeft[i]);
+        ConPrintf(StdOut, (LPWSTR)L"\n");
+    }
+
+    return TRUE;
+}
+
+BOOL HandleHelpCommand(LPWSTR szCommand, int argcLeft, LPWSTR * argvLeft)
+{
+    if (argcLeft != 0)
+    {
+        return FALSE;
+    }
+
+    ConPrintf(StdOut, (LPWSTR)L"\n");
+    ConResPuts(StdOut, IDS_APPTITLE);
+    ConPrintf(StdOut, (LPWSTR)L"\n\n");
+
+    ConResPuts(StdOut, IDS_CMD_USAGE);
+    ConPrintf(StdOut, (LPWSTR)L"%ls\n", UsageString);
+    return TRUE;
+}
+
+BOOL ParseCmdAndExecute(LPWSTR lpCmdLine, BOOL bIsFirstLaunch, int nCmdShow)
+{
+    INT argc;
+    LPWSTR *argv = CommandLineToArgvW(lpCmdLine, &argc);
+
+    if (!argv)
+    {
+        return FALSE;
+    }
+
+    if (argc == 1) // RAPPS is launched without options
+    {
+        // Close the console, and open MainWindow
+        FreeConsole();
+
+
+        // Check for if rapps MainWindow is already launched in another process
+        HANDLE hMutex;
+
+        hMutex = CreateMutexW(NULL, FALSE, szWindowClass);
+        if ((!hMutex) || (GetLastError() == ERROR_ALREADY_EXISTS))
+        {
+            /* If already started, it is found its window */
+            HWND hWindow = FindWindowW(szWindowClass, NULL);
+
+            /* Activate window */
+            ShowWindow(hWindow, SW_SHOWNORMAL);
+            SetForegroundWindow(hWindow);
+            return FALSE;
+        }
+
+        if (SettingsInfo.bUpdateAtStart || bIsFirstLaunch)
+            CAvailableApps::ForceUpdateAppsDB();
+
+        MainWindowLoop(nCmdShow);
+
+        if (hMutex)
+            CloseHandle(hMutex);
+    }
+    else if (MatchCmdOption(argv[1], CMD_KEY_INSTALL))
+    {
+        return HandleInstallCommand(argv[1], argc - 2, argv + 2);
+    }
+    else if (MatchCmdOption(argv[1], CMD_KEY_SETUP))
+    {
+        return HandleSetupCommand(argv[1], argc - 2, argv + 2);
+    }
+    else if (MatchCmdOption(argv[1], CMD_KEY_FIND))
+    {
+        return HandleFindCommand(argv[1], argc - 2, argv + 2);
+    }
+    else if (MatchCmdOption(argv[1], CMD_KEY_HELP))
+    {
+        return HandleHelpCommand(argv[1], argc - 2, argv + 2);
+    }
+    else
+    {
+        // unrecognized/invalid options
+        ConResPuts(StdOut, IDS_CMD_INVALID_OPTION);
+        ConPrintf(StdOut, (LPWSTR)L"\n");
```

Comment:
Cast now unneeded!

Reply:
By the way (same remark for the other cases), you may consider adding the newline inside the resource string instead.

Reply:
@kernelbin I do agree with adding the newline inside the resource string, other than that this PR looks good to go.

Reply:
ok! now I move all of them into resource strings!

Reply:
> Cast now unneeded!

probably I can remove them after merged into master🤣